crates/weaver_semconv/src/any_value.rs

```diff
@@ -0,0 +1,336 @@
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// An array of boolean attribute.
+    #[serde(rename = "boolean[]")]
+    Booleans {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+    },
+
+    /// The value type is a map of key, value pairs
+    Map {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec,
+        /// The collection of key, values where the value is an `AnyValueSpec`
+        fields: Vec<AnyValueSpec>,
+    },
+
+    /// The value type will just be a bytes.
+    Bytes {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec
+    },
+
+    /// The value type is not specified.
+    Undefined {
+        /// The common value specification
+        #[serde(flatten)]
+        common: AnyValueCommonSpec
+    },
+}
+
+/// The Common Value specification for properties associated with an "AnyValue", this
+/// is similar to the current `AttributeSpec` as at the proto level an Attribute
+/// is defined as a "KeyValue".
+/// While this is (currently) a duplication of the existing AttributeSpec, this is
+/// to reduce the size of the change set.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "snake_case")]
+pub struct AnyValueCommonSpec {
+    /// String that uniquely identifies the enum entry.
+    pub id: String,
+    /// A brief description of the value
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub brief: String,
+    /// A more elaborate description of the value.
+    /// It defaults to an empty string.
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub note: String,
+    /// Specifies the stability of the value.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stability: Option<Stability>,
+    /// Sequence of examples for the value or single example
+    /// value. If only a single example is provided, it can
+    /// directly be reported without encapsulating it
+    /// into a sequence/dictionary.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub examples: Option<Examples>,
+}
+
+/// Implements a human readable display for AnyValueType.
+impl Display for AnyValueSpec {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AnyValueSpec::Map { fields, .. } => {
+                let entries = fields
+                    .iter()
+                    .map(|v| format!("{}", v))
+                    .collect::<Vec<String>>()
+                    .join(", ");
+                write!(f, "map{{{}}}", entries)
+            }
+            AnyValueSpec::Bytes { .. } => write!(f, "byte[]"),
+            AnyValueSpec::Undefined { .. } => write!(f, "undefined"),
+            _ => write!(f, "{:?}", self),
```

Comment:
In the example I provided, I left this part unimplemented, but I believe we should provide a dedicated rendering for each variant.

Reply:
Yeah, I made this PR with mostly just your changes so that we could let everyone else see the changes, I've got several updates (currently just locally in my event PR) which provides all the "extra" stuff to support resolving.

Reply:
Still working on the test details -- I'd like to push the changes back into this PR by end of day -- I may (or may not) make that goal at this point